### ISSUE TITLE
Recaptcha box never loads, makes unending requests to Google

### ISSUE THREAD

**Issue by ra-external** [dozoisch/react-google-recaptcha#182]
react-google-recaptcha version: 2.1.0

I'm using the  react-google-recaptcha to integrate Google Recaptcha into a NextJS app. 

```
   
 <ReCAPTCHA
        sitekey={process.env.GOOGLE_RECAPTCHA_SITE_KEY}
        onChange={onRecaptchaClick}
        theme='dark'
        size='normal'
        ref={recaptchaRef}
        onErrored={resetCaptcha}
        onExpired={resetCaptcha}
      />


```

I'm running the app in a dev environment where the domain is of this form

`client.dev.something-dev.com
`
I've registered `something-dev.com `as one of my domains (another is `localhost`) with Google Recaptcha Admin.

The Recaptcha box works fine on `localhost`, but on the dev server, which is hosted on ECS on AWS, the Recaptcha box never fully loads. My DevOps sees a repeating error

` Uncaught DOMException: Blocked a frame with origin "https://www.google.com" from accessing a cross-origin frame.
`
but I don't see that. Instead for me, in Chrome, the Recaptcha box keeps flashing, and the console shows it's trying to load resources from google.com/recaptcha/api2and gstatic over and over again, but the requests remain pending. In Firefox and Safari I get errors like this

![image](https://user-images.githubusercontent.com/60699805/87703449-2bfbc500-c79b-11ea-9dcc-4936d1bec88b.png)

and on Firefox I get unending requests that get `200` responses

![image](https://user-images.githubusercontent.com/60699805/87705663-8c403600-c79e-11ea-8cd8-549dedc51f39.png)

WAF is turned off on the AWS server my DevOps tells me and he doesn't know what the issue might be.

Have you run into anything like this before? We have just spent a couple hours trying to debug this and gotten nowhere. Any help much appreciated!

**Comment by esanchezvz** [comment]
I'm having the same problem.
Also using Nextjs but my dev server is hosted with vercel.
I figured out that the infinite requests were happening because I was reseting the ReCAPTCHA instance whenever an errored ocurred, which would then try to load the instance again which would error out and reset again and again.
But I cannot figure out why it doesn't work on my dev server. I also registered two domains on the recaptcha admin console (mydomain.com and dev.mydomain.com) (I am using a different key for localhost, which works fine)
Were you able to find out what the error is?

**Comment by esanchezvz** [comment]
As soon as replied, I got an idea as to what may be causing the error for me.
In my case it was that the process.env.CAPTCHA_SITE_KEY was not correct. I had set it with the key I registered for localhost and not my dev server domain. As soon as I changed that the error was gone.
So, I would think that in your case, the first thing to do would be to not reset the captcha instance when it thorws an error. Then check your env variables and that the domains are registered correctly in the admin console. I don't know what else could make the captcha instace throw an error (other than bad internet connection).
Hope this helps.
